source/val/validate_decorations.cpp

```diff
@@ -514,7 +516,7 @@ spv_result_t checkLayout(uint32_t struct_id, const char* storage_class_str,
       if (SpvOpTypeStruct == arrayInst->opcode() &&
           SPV_SUCCESS != (recursive_status = checkLayout(
                               typeId, storage_class_str, decoration_str,
-                              blockRules, constraints, vstate)))
+                              blockRules, offset, constraints, vstate)))
```

Comment:
It looks like this is still only going to check the offset for the first struct in an array. I think that it can still go wrong where the ArrayStride is not a multiple of 4. For example:
```
struct {
    struct {
        float a;        // Offset 0
        vec2 b;         // Offset 4
        float c, d;     // Offsets 12, 16
    } p[2];
};
```
The size of the struct `p` is 20, and the alignment is 8, so the ArrayStride will be 24. That means that `p[1].b` will still improperly straddle from 28-36.